solr/core/src/test/org/apache/solr/search/neural/KnnQParserTest.java

```diff
@@ -205,6 +214,31 @@ public void correctVectorField_shouldSearchOnThatField() {
         "//result/doc[3]/str[@name='id'][.='12']");
   }
 
+  @Test
+  public void vectorByteEncodingField_shouldSearchOnThatField() {
+    String vectorToSearch = "[2.0, 2.0, 1.0, 3.0]";
+
+    assertQ(
+        req(CommonParams.Q, "{!knn f=vector_byte_encoding topK=2}" + vectorToSearch, "fl", "id"),
+        "//result[@numFound='2']",
+        "//result/doc[1]/str[@name='id'][.='2']",
+        "//result/doc[2]/str[@name='id'][.='3']");
+  }
+
+  @Test
+  public void vectorByteEncodingField_shouldNotApproximateQueryVectorValueToIntegerValues() {
+    String vectorToSearch = "[7.9f, 2.9f, 1.9f, 3.9f]";
+
+    // If we consider the vectors with floating values, document 7 has an higher score than document
+    // 6.
+    // If instead we take only the integer part, document 6 gets an higher score.
```

Comment:
having various if in the comment make this test difficult to read, where is this condition?